src/MemUtils_CUDA.cxx

```diff
@@ -193,6 +234,9 @@ int getCudaReductionId()
   static int first_time_called = true;
 
   if (first_time_called) {
```

Comment:
This doesn't need to be done for this PR, but put in an issue to fix some of these potential race conditions.  This should be using a std::once for example.